install.sh

```diff
@@ -1,32 +1,64 @@
-#!/bin/bash
-
-get_os () {
-  if [[ "$OSTYPE" == "linux-gnu" ]]; then
-    echo "linux"
-  elif [[ "$OSTYPE" == "freebsd"* ]]; then
-    echo "linux"
-  elif [[ "$OSTYPE" == "darwin"* ]]; then
-    echo "macos"
-  elif [[ "$OSTYPE" == "cygwin" ]]; then
-    echo "windows"
-  elif [[ "$OSTYPE" == "msys" ]]; then
-    echo "windows"
-  elif [[ "$OSTYPE" == "win32" ]]; then
-    echo "windows"
+#!/bin/sh
+
+set -eu
+
+DESTDIR="${DESTDIR:-/}"
+PREFIX="${PREFIX:-"$DESTDIR/usr/local"}"
+
+releases_url="https://github.com/soywod/himalaya/releases"
+
+uname_os() {
+  os=$(uname -s | tr '[:upper:]' '[:lower:]')
+
+  case $os in
+    msys*) os="windows" ;;
+    mingw*) os="windows" ;;
+    cygwin*) os="windows" ;;
+    win*) os="windows" ;;
+  esac
+
+  echo "$os"
+}
+
+get_system() {
+  case $(uname_os) in
+    linux | freebsd) system="linux" ;;
+    darwin) system="macos" ;;
+    windows) system="windows" ;;
+  esac
+
+  echo "$system"
+}
+
+start() {
+  system=$(get_system)
+  if [ -z "$system" ]; then
+    echo "Error: Unsupported system: $system"
+    exit 1
+  fi
+
+  if ! tmpdir=$(mktemp -d); then
+    echo "Error: Failed to create tmpdir"
+    exit 1
   else
-    return -1
+    trap 'rm -rf $tmpdir' EXIT
   fi
-}
 
-OS=`get_os`
+  echo "Downloading latest $system release…"
+  curl -sLo "$tmpdir/himalaya.tar.gz" "$releases_url/latest/download/himalaya-$system.tar.gz"
+
+  echo "Installing binary…"
+  tar -xzf "$tmpdir/himalaya.tar.gz" -C "$tmpdir"
 
-cd /tmp
-echo "Downloading latest ${OS} release…"
-curl -sLo himalaya.tar.gz "https://github.com/soywod/himalaya/releases/latest/download/himalaya-${OS}.tar.gz"
-echo "Installing binary…"
-tar -xzf himalaya.tar.gz
-rm himalaya.tar.gz
-chmod u+x himalaya.exe
-sudo mv himalaya.exe /usr/local/bin/himalaya
+  if [ -w "$PREFIX" ]; then
+    mkdir -p "$PREFIX/bin"
+    install "$tmpdir/himalaya.exe" "$PREFIX/bin/himalaya"
+  else
+    sudo mkdir -p "$PREFIX/bin"
+    sudo install "$tmpdir/himalaya.exe" "$PREFIX/bin/himalaya"
+  fi
+
+  echo "$("$PREFIX/bin/himalaya" --version) installed!"
+}
 
-echo "$(himalaya --version) installed!"
+start
```

Comment:
What's the benefit of wrapping everything in a `start` function?

Reply:
If the only benefits are for testing purposes then we can forget! I don't plan to test the `install.sh` :laughing: (except for the OS detection there is no logic). So I would remove it. [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).